giskardpy/motion_statechart/monitors/feature_monitors.py

```diff
@@ -49,10 +49,7 @@ def __post_init__(self):
         self.controlled_feature = self.tip_point
         super().__post_init__()
 
-        distance = self.root_P_controlled_feature.distance_projected_on_vector(
-            self.root_P_reference_feature,
-            cas.Vector3(0, 0, 1),
-        )
+        distance = cas.dot(self.root_P_controlled_feature - self.root_P_reference_feature, cas.Vector3(0, 0, 1))
```

Comment:
`distance = (self.root_P_controlled_feature - self.root_P_reference_feature) @ cas.Vector3.Z()`
- you should now use "@" instead of dot, i think its more readable
- vectors have factories for things like z axis